### PR TITLE
fix(server): validate WebSocket Origin header (GHSA-w3hv-x4fp-6h6j)

### PR DESCRIPTION
**Pull request by nick-pape**
## Summary
- Add `Origin` header validation to the WebSocket connection handler in `ws-bridge.ts`, preventing cross-origin WebSocket hijacking (CWE-346)
- In local mode (default), only loopback origins (`localhost`, `127.0.0.1`, `[::1]`) on the correct port are accepted; in network mode (`--allow-network`) any hostname on the correct port is allowed
- Connections with no `Origin` header (CLI/SDK clients) are always permitted; invalid origins are rejected with close code `4003`

## Test plan
- [x] Unit tests for `isAllowedOrigin()` — 13 cases covering allowed/rejected origins, port matching, `allowNetwork` mode, edge cases
- [x] Integration tests with real HTTP+WS server — verifies close code 4003 for bad origins, accepts good origins and no-origin connections
- [x] Existing `ws-bridge-send-input` tests updated and passing
- [x] Manual verification against live server: evil origin rejected (4003), correct origin accepted, no-origin accepted, wrong-port rejected (4003)
- [x] Web UI loads and shows "Connected" status with legitimate same-origin WebSocket

## Security Advisory
[GHSA-w3hv-x4fp-6h6j](https://github.com/nick-pape/grackle/security/advisories/GHSA-w3hv-x4fp-6h6j)

### CODE REVIEW

File: packages/server/src/ws-bridge-origin.test.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+describe("isAllowedOrigin", () => {
+  const WEB_PORT = 3000;
+
+  it("allows connections with no origin header", () => {
+    expect(isAllowedOrigin(undefined, WEB_PORT, false)).toBe(true);
+  });
+
+  it("allows connections with empty-string origin", () => {
+    expect(isAllowedOrigin("", WEB_PORT, false)).toBe(true);
```
[the remark this answers is not in the document]
Comment (nick-pape) [reply packages/server/src/ws-bridge-origin.test.ts]: Fixed in 49faa68 — test now expects rejection for empty-string origin.
